### DISCUSSION TITLE
Why my brightness is not controlled properly when it is higher than 50%?

### DISCUSSION THREAD

**Discussion by dev-coco** [MonitorControl/MonitorControl#1084] [Q&A] [answered]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] Looked through [the wiki](https://github.com/MonitorControl/MonitorControl/wiki)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the bug

1. The brightness control bar does not control the brightness properly. The brightness will be reduced only when the brightness is below 50%, and there will be no change in adjusting the brightness between 50% ~ 100%.
2. When adjusting the brightness, there is a probability that the display will crash, but so far I have not found any law to reproduce, it seems to be a random occurrence. I had to reboot the device to get it back to normal.

I am not sure if I have described it clearly, if necessary please leave your email and I can send a video demonstrating the bug.

### Steps to reproduce

1. Use the reduce brightness button on the keyboard to reduce the brightness
2. Reduce the brightness only when the brightness is reduced below 50%, and it has no effect when it is between 50% ~ 100%.

### Expected behavior

Any range between 0% ~ 100% should take effect normally.

### Anything else?

This bug was found in MonitorControl v4.1.0, I tried v4.0.2 and v4.0.1 both are working fine.

### Environment Information (please complete the following information)

```markdown
- macOS version: 12.3.1 Monterey
- Mac model: Mac mini (M1, 2020)
- MonitorControl version: v4.1.0
- Monitor(s): LG HDR 4K
- Apple Silicon/M1 (yes or no): yes
```

**Comment by waydabber** [comment]
Hi, the app uses a combined brightness control approach by default. Between 0-50% the dimming is achieved via software means and beyond 50% there is hardware backlight control. If this part of the scale does not work then for some reason DDC control fails.

Can you an alternative app like [BetterDisplay](https://github.com/waydabber/BetterDummy) and see if control works properly?

**Comment by waydabber** [comment, marked as answer]
Ohh but I see you did run v4.0.2 earlier with no problem. The issue most probably is that you are using HDMI connection which blocks hardware backlight control. You'd need to use the USB-C port of your Mini to connect the display for that. MC 4.1.0 had a check removed that blacklisted the HDMI port of various M1 devices (this is because many models are out now with differing service ports for HDMI and a blanket ban would also blacklist USB-C to HDMI dongles, most of which would work otherwise). If you want to fix this by using software control only, navigate to the Displays section of MonitorControl and disable DDC for the display.

Hope this helps!

**Reply by dev-coco** [reply]
It's work for me.

**Comment by dev-coco** [comment]
How about this problem? When adjusting the brightness, there is a probability that the display will crash, but so far I have not found any law to reproduce, it seems to be a random occurrence. I had to reboot the device to get it back to normal.

**Reply by waydabber** [reply]
This is also a side-effect of the Mac Mini HDMI problem as the converter chip in 2018+ macs garble ddc i2c communication. I did experience the same with an LG UHD display. You should disable DDC or use USB-C if you want to have hardware control over the backlight via DDC.

**Reply by dev-coco** [reply]
Ok, thanks a lot.
